packages/analytics/amplify_analytics_pinpoint/lib/src/flutter_path_provider/flutter_path_provider_html.dart

```diff
@@ -0,0 +1,24 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:amplify_analytics_pinpoint_dart/amplify_analytics_pinpoint_dart.dart';
+
+/// Storage location is not needed on Web
+/// As we used IndexedDB
+class FlutterPathProvider extends CachedEventsPathProvider {
+  @override
+  Future<String> getApplicationSupportPath() async {
+    throw UnsupportedError('Do not call FlutterPathProvider on web');
```

Comment:
Do you need to split this out if it will just throw unimplemented? I imagine there are separate code paths from web and io where this is used.

If you are doing it just to maintain a consistent pattern, I think that is fine. But I am not sure you really need to.